### ISSUE TITLE
piper won't start

### ISSUE THREAD

**Issue by muzena** [libratbag/piper#66]
In Ubuntu Artful i get this error in terminal:

$ piper
Traceback (most recent call last):
  File "/usr/bin/piper", line 32, in <module>
    from piper.application import Application
ModuleNotFoundError: No module named 'piper'

**Comment by whot** [comment]
sounds like #15, but that should've been fixed. Are you running git master?

**Comment by muzena** [comment]
@whot 
Yes

**Comment by Hjdskes** [comment]
Just to make sure that this is an issue on Piper's side, can you run the following?

```sh
git clone https://github.com/libratbag/piper
cd piper
meson --prefix=/usr build
ninja -C build
sudo ninja -C build install
```

If things still do not work, can you provide `build/meson-logs/install-log.txt` here?

**Comment by muzena** [comment]
Same thing, same error.
[install-log.txt](https://github.com/libratbag/piper/files/1185882/install-log.txt)
[meson-log.txt](https://github.com/libratbag/piper/files/1185883/meson-log.txt)

**Comment by cr1ogen** [comment]
I have the exact problem, Debian Sid

**Comment by Hjdskes** [comment]
Can we get the output of `python3 -c "import sys; print('\n'.join(sys.path))"` and `python3 -c "import sysconfig; print(sysconfig.get_path('purelib'));"`?

Does Piper work if you run it with `PYTHONPATH=$PYTHONPATH:/usr/lib/python3.6/site-packages piper`?

**Comment by muzena** [comment]
$ python3 -c "import sys; print('\n'.join(sys.path))"
/usr/lib/python36.zip
/usr/lib/python3.6
/usr/lib/python3.6/lib-dynload
/usr/local/lib/python3.6/dist-packages
/usr/lib/python3/dist-packages
/usr/lib/python3.6/dist-packages

$ python3 -c "import sysconfig; print(sysconfig.get_path('purelib'));" 
/usr/lib/python3.6/site-packages

$ PYTHONPATH=$PYTHONPATH:/usr/lib/python3.6/site-packages piper
Traceback (most recent call last):
  File "/usr/bin/piper", line 32, in <module>
    from piper.application import Application
  File "/home/sinisa/Dokumenti/piper/piper/piper/application.py", line 18, in <module>
    from .window import Window
  File "/home/sinisa/Dokumenti/piper/piper/piper/window.py", line 21, in <module>
    from .resolutionspage import ResolutionsPage
  File "/home/sinisa/Dokumenti/piper/piper/piper/resolutionspage.py", line 20, in <module>
    from .mousemap import MouseMap
  File "/home/sinisa/Dokumenti/piper/piper/piper/mousemap.py", line 21, in <module>
    from lxml import etree
ModuleNotFoundError: No module named 'lxml'

**Comment by Hjdskes** [comment]
This bug isn't with Piper; your Python isn't configured correctly.

@whot What do we do with this? There are now three users on three different APT-based distros (@muzena on Ubuntu Artful, @cr1ogen on Debian Sid and @Chronigan on Linux Mint (see #15)) who report this issue.

**Comment by muzena** [comment]
I installed python3-lxml, but piper, don't work.

**Comment by Hjdskes** [comment]
You'll have to give more details than that. What is printed when you try running Piper now?

**Comment by whot** [comment]
well, I manged to verify this with a new ubuntu 17.10 alpha image. Installing piper with prefix /usr will install it into site-packages, but  that isn't in the python path by default. We don't do anything in piper here that should trigger this issue. meson just gets purelib, so this still looks like a system configuration issue.

it's beyond my python knowledge and certainly beyond my available time budget, sorry.

**Comment by bentiss** [comment]
I just pinged the debian maintainer of libratbag. And basically, this will be fixed once we are releasing the next stable version, but until them, we are not going to look at it because it is a distribution issue.

The current version of libratbag in debian is old, but it should be working for most of the features. The piper UI is currently under heavy rework and honestly you might not want to use it right now. We are fixing bugs on a everyday basis, and this would be a pre-alpha that you would be running.

Basically, sorry, you will have to wait a little bit more before being able to use the latest version.

If you **really** want to test the pre-alpha, do not install anything, clone `libratbag`, install it, and in the `piper` build directory, run `piper.devel` without installing it.

**Comment by muzena** [comment]
@Hjdskes 
Terminal output is same as before python3-lxml.
I think that something is wrong with python path or path is different in ubuntu /usr/lib    /usr/local/lib

Does ratbag daemon must to be started automatically after power on PC?

**Comment by whot** [comment]
ratbagd is systemd dbus-activated, so it should start once piper tries to connect to it. But this error happens before it connects to ratbagd during initial python module load, it's an issue completely independent of libratbag

**Comment by Hjdskes** [comment]
> Terminal output is same as before python3-lxml.
I think that something is wrong with python path or path is different in ubuntu /usr/lib /usr/local/lib

Installing python3-lxml isn't mean to magically solve your issue. You'll still need to run `PYTHONPATH=$PYTHONPATH:/usr/lib/python3.6/site-packages piper`, but you won't get `ModuleNotFoundError: No module named 'lxml'` anymore.

**Comment by kungpaogao** [comment]
**Disclaimer:** I'm a newb.

I was about to make a new issue before I found this. I'm also still experiencing similar issues where piper does not run. Is there any update on this? 

Also, I tried installing and running ratbag, but that also has issues where the command does not close. Is that normal? Does it have anything to do with the issues in running piper?

- Running `sudo ratbagd` returns a new line and nothing else. 
- Running `sudo ratbagd --verbose` returns:
 ```ratbag debug: Logitech M705 is device '/dev/hidraw4'.
ratbag debug: report ID 02
ratbag debug: report ID 10
ratbag debug: report ID 11
ratbag debug: report ID 20
ratbag debug: report ID 21
ratbag debug: no profile type given
ratbag debug: driver match found: Logitech HID++1.0
event9: "Logitech M705", 1 profiles
```
- Installing ratbagd from apt and then running `sudo ratbagd` returns `ratbagd error: Failed: File exists`
- Starting ratbagd using systemctl doesn't do anything either.

I'm running Xubuntu 17.10.  I have used `meson` and `ninja` to install everything as specified in the README. And running the python commands return normal values.
